tests/Core/Ruleset/PopulateTokenListenersTest.php

```diff
@@ -0,0 +1,548 @@
+                    sprintf('Unexpected value for "class" key for sniff class %s for token %s', $className, Tokens::tokenName($token))
+                );
+
+                $this->assertArrayHasKey(
+                    'source',
+                    $details,
+                    sprintf('"source" key missing for sniff class %s for token %s', $className, Tokens::tokenName($token))
+                );
+
+                $this->assertTrue(
+                    is_string($details['source']),
+                    sprintf('Value for "source" key is not a string for token %s', Tokens::tokenName($token))
+                );
+
+                $expected = '.'.substr($className, (strrpos($className, '\\') + 1), -5);
+
+                $this->assertStringEndsWith(
+                    $expected,
+                    $details['source'],
+                    sprintf('Unexpected value for "source" key for sniff class %s for token %s', $className, Tokens::tokenName($token))
+                );
+            }//end foreach
+        }//end foreach
+
+    }//end testSetsClassAndSourceIndexes()
+
+
+    /**
+     * Verifies that sniffs by default are listening for PHP files only.
+     *
+     * @return void
+     */
+    public function testSetsSupportedTokenizersToPHPByDefault()
+    {
+        $exclude  = 'PHP_CodeSniffer\\Standards\\Generic\\Sniffs\\Files\\EndFileNewlineSniff';
+        $expected = ['PHP' => 'PHP'];
+
+        foreach (self::$ruleset->tokenListeners as $token => $listeners) {
+            $this->assertTrue(is_array($listeners), 'No listeners registered for token'.Tokens::tokenName($token));
+
+            foreach ($listeners as $className => $details) {
+                if ($className === $exclude) {
+                    // Skip this one as it is the one sniff for which things will be different.
+                    continue;
+                }
+
+                $this->assertArrayHasKey(
+                    'tokenizers',
+                    $details,
+                    sprintf('"tokenizers" key missing for sniff class %s for token %s', $className, Tokens::tokenName($token))
+                );
+
+                $this->assertSame(
+                    $expected,
+                    $details['tokenizers'],
+                    sprintf('Unexpected value for "tokenizers" key for sniff class %s for token %s', $className, Tokens::tokenName($token))
+                );
+            }
+        }//end foreach
+
+    }//end testSetsSupportedTokenizersToPHPByDefault()
+
+
+    /**
+     * Test that if a sniff has the $supportedTokenizers property set, the tokenizers listed there
+     * will be registered in the listeners array.
+     *
+     * @param int $token The token constant for which the sniff should be registered.
+     *
+     * @dataProvider dataSetsSupportedTokenizersWhenProvidedBySniff
+     *
+     * @return void
+     */
+    public function testSetsSupportedTokenizersWhenProvidedBySniff($token)
```

Comment:
No, not really on my radar and as you correctly say: outside the scope of this PR. This PR is about adding _tests for the existing code_, not about _improving_ the existing code.

Having said that, I'm not all that inclined to add an `is_array()` check for the following reasons:
1. As of PHPCS 4.0, the property is no longer supported as there will only be one supported tokenizer.
2. Guard code IMO should be mostly about protecting end users from accidental mistakes. Sniff developers should have at least a basic understanding of what they're doing.
3. The property has always been an array property, there is no new of changed behaviour (other than the upcoming change in 4.0).
4. If a sniff would do this wrong, I believe it is a good thing that this would lead to a PHP error during development of the sniff.

Reply:
Thanks for taking the time to elaborate on why you are not inclined to add an `is_array()` check. Makes sense to me.